src/test/java/core/basesyntax/service/strategy/ReduceQuantityStrategyTest.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax.service.strategy;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import core.basesyntax.dao.FruitDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.dto.FruitDto;
+import core.basesyntax.model.Fruit;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ReduceQuantityStrategyTest {
+    private static OperationStrategy strategy;
+
+    @BeforeClass
+    public static void beforeClass() {
+        strategy = new ReduceQuantityStrategy(new FruitDaoImpl());
+    }
+
+    @Test
+    public void test_reduceProcess_Ok() {
+        Fruit fruit = new Fruit("cherry");
+        Storage.fruitStorage.put(fruit, 25000);
+        strategy.process(new FruitDto("p", "cherry", 77));
+        Integer expected = 24923;
+        Integer actual = Storage.fruitStorage.get(fruit);
+        assertEquals(expected, actual);
+        Storage.fruitStorage.clear();
+    }
+
+    @Test
+    public void test_reduceProcess_NotOk() {
+        try {
+            Fruit fruit = new Fruit("cherry");
+            Storage.fruitStorage.put(fruit, 25000);
+            strategy.process(new FruitDto("p", "cherry", 25077));
```

Comment:
Можливо краще винести "p" в константу(можливо).
краще використати метод з fruit, тому що не відомо до чого тут той string, а fruit.getName() одразу зрозуміло (ти його поклав в Storage, і так само до нього відносишся)
```suggestion
            strategy.process(new FruitDto(PURCHASE, fruit.getName(), 25077));
```